telegram/ext/filters.py

```diff
@@ -603,6 +603,41 @@ def filter(self, message: Message) -> bool:
         xml = mime_type('application/xml')
         zip = mime_type('application/zip')
 
+        class file_extension(MessageFilter):
+            """This filter filters documents by their file ending/extension.
+
+            Note:
+                This Filter only filters by the file ending/extension of the document,
+                    it doesn't check the validity of document.
+                The user can manipulate the file extension of a document and
+                    send media with wrong types that don't fit to this handler.
+
+            Example:
+                ``Filters.document.file_extension('jpg')`` filters files with extension ``.jpg``.
+            """
+
+            def __init__(self, file_extension: Optional[str]):
+                """Initialize the extension you want to filter.
+
+                Args:
+                    file_extension (str, optional): media file extension you want to filter.
+                """
+                if file_extension is None:
+                    self.file_extension = file_extension
+                elif file_extension.startswith("."):
+                    self.file_extension = file_extension
+                else:
+                    self.file_extension = f".{file_extension}"
```

Comment:
my point was that this should only be either

* `self.file_extension = file_extension` and the docs should nota that the user needs to care abot the dot
* `self.file_extension = f'.{file_extension}'` and the docs should note that a dot will be added, not matter what